contracts/math/SafeMath.sol

```diff
@@ -45,4 +45,17 @@ library SafeMath {
     assert(c >= a);
     return c;
   }
+
+  /**
+  * @dev Integer modulo operation of two numbers.
+  */
+  function mod(uint256 a, uint256 b) internal pure returns (uint256 c) {
+    // d = a / b
+    // a = d * b + c
+    // c = a - d * b    
+    uint256 d = div(a,b);
+    c = sub(a, mul(d,b));
+    assert(c < b);
```

Comment:
:+1:, this is in line with what's being discussed at #1120 (unless I'm mistaken @leonardoalt?)

Reply:
@nventuro Yes! (if the other operations add the necessary requires)

Reply:
It's more fundamental than that, by definition the result of a modulo operation will always be strictly less than the divisor (it's the remainder of dividing by it), which is why I thought this was an interesting example of 'correct' `assert` usage.

Reply:
Sure, but that's exactly the point. Since here `mod` is implemented using the other operators, if they overflow etc the assertion might break. By having the other operations ensure their correctness, the corrected of `mod` follows.

Reply:
And you are totally right in saying this is a nice example of the usage of assert: you have the pre-conditions that are true from the other operations, which imply the `mod` property.